Add published_dates_with_notification_button partial

sairamya93 opened the pull request:
, [Jira issue PNP-6369](https://gov-uk.atlassian.net/browse/PNP-6369)⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Add `published_dates_with_notification_button` partial

## Why

As part of app consolidation, `published_dates_with_notification button` partial is required for the document type `detailed_guide`. The partial would also be needed for other document types - `call_for_evidence`, `consultation` and `publication`.

Trello card: https://trello.com/c/XJekbr1s

## How

Based on the code in government-frontend where the pages are rendered from currently.

deborahchua committed:
Add published_dates_with_notification_button partial

 - Copied over the partial from government-frontend that renders pages with notification button along with the published dates
- Currently in frontend, the @has_govuk_account value is not being set. So here we would pass the value to the partial from the view that includes this partial and it will be false when the value is not present.

Audit trail for shared partial: https://github.com/alphagov/government-frontend/blob/af1c374341d9b2c12e05d79a95293ccd1e892e3f/app/views/shared/_published_dates_with_notification_button.html.erb
Audit trail for the related stylesheet: https://github.com/alphagov/government-frontend/blob/af1c374341d9b2c12e05d79a95293ccd1e892e3f/app/assets/stylesheets/views/_published-dates-button-group.scss
Config for scss: https://github.com/alphagov/government-frontend/blob/af1c374341d9b2c12e05d79a95293ccd1e892e3f/config/initializers/dartsass.rb#L6

KludgeKML commented on app/views/shared/_published_dates_with_notification_button.html.erb:
```diff
@@ -0,0 +1,40 @@
+  </h2>
+
+  <%= render "govuk_publishing_components/components/single_page_notification_button", {
+    base_path: @content_item.base_path,
```
Could lose the @ from content_item? (here and below)

KludgeKML commented on app/views/shared/_published_dates_with_notification_button.html.erb:
```diff
@@ -0,0 +1,40 @@
+
+  <%= render "govuk_publishing_components/components/single_page_notification_button", {
+    base_path: @content_item.base_path,
+    js_enhancement: @has_govuk_account,
```
Do we know what this does?

KludgeKML replied:
Okay for the moment, but we probably need to look into this variable (the only other place it's used in frontend is in another shared template, and it's never set. So I think we should ultimately make this use and the other locals to the partials which can be set when called)

sairamya93 replied:
I have updated the partial to set the value for the `account` from the view that renders this partial else defaults to false.

deborahchua commented on app/views/shared/_published_dates_with_notification_button.html.erb:
Can we update the audit trail links in the commit description? I think this file has changed slightly since I made the commit you've cherrypicked.

deborahchua commented on spec/views/shared/_published_dates_with_notification_button.html.erb_spec.rb:
```diff
@@ -0,0 +1,43 @@
+  end
+end
+
+# To remove this once DetailedGuide PR is made
```
Is this meant to be included as part of the PR?

sairamya93 replied:
Not exactly. I had added in a dummy class here to handle the tests for the partial as we don't yet have the specific document type views like `detailed_guide`  that actually render this partial. Once we have the detailed_guide view, we could remove the dummy class.

KludgeKML commented on config/locales/ar.yml:
```diff
@@ -96,10 +97,12 @@ ar:
     today:
     united-kingdom_slug:
     upcoming_bank_holidays:
+    visit: 'تفضّل بزيارة:'
```
It's possible this command was a bit too broad - it's brought in this common.visit, which isn't used in this partial. Perhaps run the command again twice, once for common.print_link, once for common.email_and_print_link, so it doesn't bring in other things?

sairamya93 committed:
Add locale keys for published_dates_with_notification_button partial

- Ran the commands: govuk-docker-run bundle exec rake "consolidation:copy_translation[common.email_and_print_link]"
govuk-docker-run bundle exec rake "consolidation:copy_translation[common.print_link]"

KludgeKML reviewed (approved):
LGTM
